src/Microsoft.AspNet.SignalR.Client/Http/DefaultHttpClient.cs

```diff
@@ -86,7 +86,11 @@ public Task<IResponse> Get(string url, Action<IRequest> prepareRequest, bool isL
                      }
                      else
                      {
-                         throw new HttpClientException(responseMessage);
+                         // Dispose the response (https://github.com/SignalR/SignalR/issues/4092)
+                         var message = responseMessage.ToString();
+                         responseMessage.RequestMessage.Dispose();
+                         responseMessage.Dispose();
+                         throw new HttpClientException(message);
```

Comment:
Why are we passing in a string here?

Reply:
So the client can see the failed response details if desired. The string was being used for the Exception message before. The only difference is instead of ToString() being called [here](https://github.com/SignalR/SignalR/blob/cf945d219a52c8103bde8bd5c6dc79f1ee0e64da/src/Microsoft.AspNet.SignalR.Client/Infrastructure/HttpClientException.cs#L46) like before, it's called before the HttpResponseMessage is disposed. Technically, I think ToString() still works after disposal, but it feels safer to just call ToString() earlier.

Reply:
What @halter73 said :)